torchvision/transforms/transforms.py

```diff
@@ -738,17 +744,30 @@ def __call__(self, tensor):
             raise ValueError("tensor and transformation matrix have incompatible shape." +
                              "[{} x {} x {}] != ".format(*tensor.size()) +
                              "{}".format(self.transformation_matrix.size(0)))
-        flat_tensor = tensor.view(1, -1)
+        flat_tensor = tensor.view(1, -1) - self.mean_vector
         transformed_tensor = torch.mm(flat_tensor, self.transformation_matrix)
         tensor = transformed_tensor.view(tensor.size())
         return tensor
 
     def __repr__(self):
-        format_string = self.__class__.__name__ + '('
+        format_string = self.__class__.__name__ + '(transformation_matrix='
         format_string += (str(self.transformation_matrix.numpy().tolist()) + ')')
+        format_string += (", (mean_vector=" + str(self.mean_vector.numpy().tolist()) + ')')
         return format_string
 
 
+class LinearTransformation(AffineTransformation):
+    """
+    Note: This transform is deprecated in favor of AffineTransformation.
+    """
+
+    def __init__(self, transformation_matrix):
+        warnings.warn("The use of the transforms.LinearTransformation transform is deprecated, " +
+                      "please use transforms.AffineTransformation instead.")
+        super(LinearTransformation, self).__init__(transformation_matrix, torch.zeros(
```

Comment:
This will fail if `transformation_matrix` is not of type `float32`, because by default `torch.zeros` returns a `float32` tensor. Can you use instead `zeros_like`? Also, why do you need to make it a `1 x D` tensor? Letting it be a `D` tensor is enough for broadcasting, and in my opinion nicer as well

Reply:
Done!